scripts/pylib/pytest-twister-harness/src/twister_harness/device/fifo_handler.py

```diff
@@ -100,14 +104,23 @@ def disconnect(self) -> None:
 
         if self._fifo_out_file:
             self._fifo_out_file.close()
+            self._fifo_out_file = None
         if self._fifo_in_file:
             self._fifo_in_file.close()
+            self._fifo_in_file = None
 
+    def cleanup(self):
+        """Clean up FIFO files when object is destroyed"""
         if os.path.exists(self._fifo_out_path):
             os.unlink(self._fifo_out_path)
         if os.path.exists(self._fifo_in_path):
             os.unlink(self._fifo_in_path)
 
+    def __del__(self):
```

Comment:
This won't work. There is no guarantee that `__del__` method will be called by garbage collector.